### PR TITLE
Apply fixes for recent upstream moto changes

### PR DESCRIPTION
**Pull request by whummer**
Apply fixes for recent upstream moto changes. In particular, the creation of model backends has changed, as well as the logic around default/main associations in route tables [here](https://github.com/spulec/moto/pull/4726). 

Changes were not yet fully verified against AWS, but parity of the EC2 changes has been verified against AWS as  per [this PR comment]([here](https://github.com/spulec/moto/pull/4726)).

### CODE REVIEW

File: localstack/services/ec2/ec2_starter.py [hunk 1]
```diff
@@ -21,30 +22,11 @@
 
 
 def patch_ec2():
-    def patch_revoke_security_group_egress(backend):
```
Comment (whummer) [inline localstack/services/ec2/ec2_starter.py:24]: Patch is obsolete, no longer required.

File: localstack/services/cloudformation/models/ec2.py [hunk 2]
```diff
@@ -339,6 +339,9 @@ def _pre_delete(resource_id, resources, resource_type, func, stack_name):
                 )
                 for rt in resp["RouteTables"]:
                     for assoc in rt.get("Associations", []):
+                        # skipping Main association (accommodating recent upstream change)
```
Comment (alexrashed) [inline localstack/services/cloudformation/models/ec2.py:342]: Just a side-node, no need to change it here: The term "recent" might be confusing at a later point in time.

**Review by alexrashed — APPROVED** [review]
This looks great! Cool to see new features, like the patching decorators, in action. And great that you already refactored the test code to be region-aware.

File: localstack/services/cloudformation/models/ec2.py (hunk 2 above)
Comment (whummer) [reply localstack/services/cloudformation/models/ec2.py:342]: Yep, great point - thanks for mentioning that @alexrashed (will include in next iteration..)

**Pull request merged**